hook.js

```diff
@@ -121,23 +195,53 @@ function createHook (meta) {
 
   const iitmURL = new URL('lib/register.js', meta.url).toString()
   async function getSource (url, context, parentGetSource) {
+    const imports = []
+    const namespaceIds = []
+
     if (hasIitm(url)) {
       const realUrl = deleteIitm(url)
       const exportNames = await getExports(realUrl, context, parentGetSource)
+      const setters = []
```

Comment:
This whole block seems to be repeated in `processStarExport()`. Can this be DRYed? There's enough happening here that could get modified in the future, so I'd have this logic only written once.

Reply:
It was a bit tricky, but I managed it. Let me know what you think.